### DISCUSSION TITLE
el-image组件的src属性未经过vite进行变基，有哪些方式可以解决呢？

### DISCUSSION THREAD

**Discussion by xhc-code** [element-plus/element-plus#12849] [Q&A]
el-image组件的src属性未经过vite进行变基，导致了引用的本地图片路径不正确。

目前已知的解决方法1：
``` js
<el-image :src="require('@/a/b/c/logo.jpg')" />
```
以上方式只能用在commonjs构建环境下（同时也是唯一的一种方式）。

第二种解决方法2（适用方法1）：
```js
import logo from '@/a/b/c/logo.jpg'
// 将这个logo放在vue响应式的变量中，之后再el-image的src中引用。
```

ESM模块构建环境的本地图片还有其他方式来引用吗？
第三方组件中的例如src或者其他本地资源，如何通过路径变基呢？

**Comment by chenxch** [comment]
https://vitejs.dev/guide/assets.html#new-url-url-import-meta-url
尝试一下这个方案

**Reply by chenxch** [reply]
https://stackoverflow.com/questions/66419471/vue-3-vite-dynamic-image-src

**Reply by xhc-code** [reply]
> https://vitejs.dev/guide/assets.html#new-url-url-import-meta-url 尝试一下这个方案

我觉得这个方案更不用考虑了，而且不兼容ssr，import的方式虽然多了几步，但也是比这个new URL方式简便。感谢你提供的方案。
